### ISSUE TITLE
Nvim: blank block always appended to command line cursor

### ISSUE THREAD

**Issue by mreed1** [junegunn/vim-oblique#51]
Hi,

I'm not sure if this is inherently a problem with Nvim, or if oblique relied on a vim only hack which has been removed (or something else), but here's my issue. (from https://github.com/neovim/neovim/pull/1820#issuecomment-74172835):

nvimrc used:
```
call plug#begin()
Plug 'tpope/vim-sensible'
Plug 'junegunn/vim-pseudocl'
Plug 'junegunn/vim-oblique'
call plug#end()
```

The issue happens with or without `vim-sensible` (note the command prompt):

with oblique:
![with](https://cloud.githubusercontent.com/assets/4174345/6178795/7dc3c606-b2e0-11e4-9762-865c3b0a7d85.png)

without:
![without](https://cloud.githubusercontent.com/assets/4174345/6178794/78f88968-b2e0-11e4-91e5-8a08a368c68f.png)

The same thing happens in xterm without tmux.

(end of quote) I thought that maybe some color sequence was being passed to stdout, and nvim might be interpreting it differently than vim.

**Comment by junegunn** [comment]
As it's not reproducible on Vim, it's probably an issue of Neovim which is constantly changing. So I won't look into it for the moment.

By the way, there is a simpler alternative to Oblique you might want to take a look.

https://github.com/pgdouyon/vim-evanesco

**Comment by Pyrohh** [comment]
> As it's not reproducible on Vim, it's probably an issue of Neovim which is constantly changing. So I won't look into it for the moment.

That's fine, but do you mind if I keep this open? Presumably there aren't many plugins that change the appearance of the command/search bar, but regardless this is the only plugin which has the issue.

> https://github.com/pgdouyon/vim-evanesco

Thanks for the reference.

**Comment by junegunn** [comment]
No problem.

**Comment by blueyed** [comment]
Another issue in this context: using `<S-Insert>` to copy from the "mouse" clipboard adds an `i` at the beginning, i.e. pasting "foo" after `/` will result in a search for "ifoo".

**Comment by junegunn** [comment]
@blueyed Nice find. Seems like a bug of neovim, it's easily reproducible.

```vim
function! s:gets()
  let s = ''
  let c = getchar()
  while 1
    if !c
      break
    endif
    let s .= nr2char(c)
    let c = getchar(0)
  endwhile
  echom s
endfunction
command! Gets call s:gets()
```

**Comment by junegunn** [comment]
https://github.com/neovim/neovim/commit/3baba1e7bc6698e6bc9f1d37fce88b30d6274bc9

`t_ve` option which this plugin uses to hide the original cursor is removed in neovim.

**Comment by rainerborene** [comment]
ping @tarruda

**Comment by tarruda** [comment]
> t_ve option which this plugin uses to hide the original cursor is removed in neovim.

I'm currently working on this. Soon nvim TUI will have the terminal codes customizable.

**Comment by blueyed** [comment]
@mreed1 
Why did you close it?  Is it fixed for you?

**Comment by mreed1** [comment]
@blueyed nah, I was just cleaning up a bunch of issues associated with my account; I'll reopen this one.
